pkg/controller/infinispan/resources/config/config_controller.go

```diff
@@ -74,6 +75,16 @@ func Add(mgr manager.Manager) error {
 }
 
 func (c *configResource) Process() (reconcile.Result, error) {
+	// Don't update the ConfigMap if an update will be required
+	if req, err := ispnctrl.IsUpgradeRequired(c.infinispan); req || err != nil {
+		return reconcile.Result{RequeueAfter: consts.DefaultWaitOnCreateResource}, nil
+	}
+
+	// If an upgrade is in progress wait for it to complete before updating the ConfigMap
+	if c.infinispan.IsAnyCondititionTrue(ispnv1.ConditionUpgrade, ispnv1.ConditionStopping) {
```

Comment:
When is ConditionUpgrade removed or set to false? Is that when podCount == 0 or the when the cluster was upgraded with new image?

Reply:
It's set to false once `!ConditionStopping` (podCount == 0) and the old StatefulSet has been deleted.